extensions/healthie/validation/createTask.zod.ts

```diff
@@ -0,0 +1,69 @@
+import { isNil } from 'lodash'
+import { z } from 'zod'
+import { stringDate } from './generic.zod'
+
+const intervalTypeEnum = z.enum(['daily', 'weekly', 'once'])
+const intervalValueWeeklyEnum = z.enum([
+  'monday',
+  'tuesday',
+  'wednesday',
+  'thursday',
+  'friday',
+  'saturday',
+  'sunday',
+])
+
+const reminderSchema = z
+  .discriminatedUnion('reminderIntervalType', [
+    z.object({
+      reminderIntervalType: z.literal(undefined),
+      reminderIntervalValue: z.literal(undefined),
+      isReminderEnabled: z.union([z.literal(false), z.literal(undefined)]),
+      reminderTime: z.literal(undefined),
+    }),
+    z.object({
+      reminderIntervalType: z.literal(intervalTypeEnum.enum.daily),
+      reminderIntervalValue: z.literal(undefined),
+      isReminderEnabled: z.literal(true),
+      reminderTime: z.coerce.number(),
+    }),
+    z.object({
+      reminderIntervalType: z.literal(intervalTypeEnum.enum.weekly),
+      reminderIntervalValue: intervalValueWeeklyEnum,
+      isReminderEnabled: z.literal(true),
+      reminderTime: z.coerce.number(),
+    }),
+    z.object({
+      reminderIntervalType: z.literal(intervalTypeEnum.enum.once),
+      reminderIntervalValue: stringDate,
+      isReminderEnabled: z.literal(true),
+      reminderTime: z.coerce.number(),
+    }),
+  ])
```

Comment:
I would add some comments here as this is a bit complex to understand for an outsider.

```typescript
// See https://docs.gethealthie.com/schema/taskreminderinput.doc.html for reference
const reminderSchema = z
  .discriminatedUnion('reminderIntervalType', [
      /**
        * If `isReminderEnabled` is false or undefined, 
        * then all other reminder properties are obsolete.
        */
      z.object({
        isReminderEnabled: z.union([z.literal(false), z.literal(undefined)]),
        reminderIntervalType: z.literal(undefined),
        reminderIntervalValue: z.literal(undefined),
        reminderTime: z.literal(undefined),
      }),
      /**
        * If `isReminderEnabled` is true, 
        * and `reminderIntervalType` is 'daily'
        * then `reminderIntervalValue` is obsolete
        */
      z.object({
        isReminderEnabled: z.literal(true),
        reminderIntervalType: z.literal(intervalTypeEnum.enum.daily),
        reminderIntervalValue: z.literal(undefined),
        reminderTime: z.coerce.number(),
      }),
      /**
        * If `isReminderEnabled` is true, 
        * and `reminderIntervalType` is 'weekly'
        * then `reminderIntervalValue` should be a comma-separated string of days of the week
        */
      z.object({
        isReminderEnabled: z.literal(true),
        reminderIntervalType: z.literal(intervalTypeEnum.enum.weekly),
        reminderIntervalValue: intervalValueWeeklyEnum,
        reminderTime: z.coerce.number(),
      }),
      /**
        * If `isReminderEnabled` is true, 
        * and `reminderIntervalType` is 'once'
        * then `reminderIntervalValue` should be an ISO8601 date
        */
      z.object({
        isReminderEnabled: z.literal(true),
        reminderIntervalType: z.literal(intervalTypeEnum.enum.once),
        reminderIntervalValue: stringDate,
        reminderTime: z.coerce.number(),
      }),
    ])
```